elliptic-curve/src/hash2field/expand_msg_xmd.rs

```diff
@@ -53,23 +54,28 @@ where
 }
 
 /// ExpandMsgXmd implements expand_message_xmd for the ExpandMsg trait
-impl<HashT> ExpandMsg for ExpandMsgXmd<HashT>
+impl<HashT, L> ExpandMsg<L> for ExpandMsgXmd<HashT>
 where
     HashT: Digest + BlockInput,
     HashT::OutputSize: IsLess<U256>,
     HashT::OutputSize: IsLessOrEqual<HashT::BlockSize>,
+    L: ArrayLength<u8> + IsLess<U65536>,
 {
-    fn expand_message(msg: &[u8], dst: &'static [u8], len_in_bytes: usize) -> Self {
+    fn expand_message(msg: &[u8], dst: &'static [u8]) -> Self {
         let b_in_bytes = HashT::OutputSize::to_usize();
-        let ell = (len_in_bytes + b_in_bytes - 1) / b_in_bytes;
+        let ell = (L::to_usize() + b_in_bytes - 1) / b_in_bytes;
         // if ell > 255 {
         //     panic!("ell was too big in expand_message_xmd");
```

Comment:
What about this?

Reply:
That's what I'm trying to figure out at the moment. It could actually be made very easy by just reducing the maximum `len_in_bytes` to 64.772, which would make it impossible to reach 255. In any case, still playing around with it.

Reply:
Alright, I re-read the spec a lot and couldn't figure out a clean way, I will just make the calculation on the type and call it a day, coming in shortly.

Reply:
This should be correct now.